js/ai/src/resource.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Action,
+  ActionContext,
+  defineAction,
+  GenkitError,
+  z,
+} from '@genkit-ai/core';
+import { Registry } from '@genkit-ai/core/registry';
+import uriTemplate from 'uri-templates';
+import { Part, PartSchema } from './model.js';
+
+/**
+ * Options for defining a resource.
+ */
+export interface ResourceOptions {
+  /**
+   * The URI of the resource. Can contain template variables.
+   */
+  uri: string;
```

Comment:
I wanted to differentiate between template and non-template resources because MCP does. I want this to map super-easily to MCP and unless we have a way to detect whether a URI does or doesn't have template params I'd rather have `uri` option for one and `template` option for the other -- it can resolve to the same kind of action etc. I just want to be able to trivially provide to MCP.

Reply:
Done.